packages/code-du-travail-nlp/api/sem_search.py

```diff
@@ -0,0 +1,146 @@
+import json
+import tensorflow as tf
+import tensorflow_hub as hub
+import numpy as np
+import tf_sentencepiece
+import unidecode
+
+from typing import List
+
+content_path = "./data/content.json"
+stops_path = "./data/stops.txt"
+
+cache_path = "~/Downloads"
+
+stringvec = List[str]
+
+
+def add_slash(c):
+    c["slug"] = "/{}".format(c["slug"])
+    return c
+
+
+class SemSearch():
+
+    def __init__(self, content_path: str, stops_path: str):
+        print("loading data...")
+        with open(content_path, "r") as f:
+            content = list(filter(
+                lambda row: "text" in row and not "code_du_travail" in row.get('slug'), json.load(f)))
+        content = list(map(add_slash, content))
+
+        with open(stops_path, "r") as f:
+            stops = f.read().splitlines()
+
+        # make a hashtable for performance
+        self.stops = {self.strip_accents(k): "" for k in stops}
```

Comment:
En général, les fonctions qu'on utilise dans une boucle, on essaie des les avoir dans une variable locale, pour eviter de chercher l'attribut a chaque fois. Ici la fonction `self.strip_accents` pourrait etre mise dans une variable temporaire `func = self.strip_accents`
Ici je crois que ca n'a pas d'importance, parce que c'est à l'init donc ce bout de code ne doit pas etre tres utilisé.
```suggestion
        func = self.strip_accents
        # self.stops = {func(k): "" for k in stops}
        # plus fun encore:
        self.stops = dict.fromkeys(map(func, stops), "")
```

Reply:
oui, on pourrait même utliser un set je pense, mais c'est vraiment secondaire.